stratisd_proc_macros/src/lib.rs

```diff
@@ -193,59 +193,45 @@ fn wrap_method(f: &mut ImplItemMethod) {
 /// Get the pool available action state level at which a pool operation ceases to be
 /// accepted.
 fn get_attr_level(attrs: &mut Vec<Attribute>) -> Option<Ident> {
-    let mut return_value = None;
-    let mut index = None;
-    for (i, attr) in attrs.iter().enumerate() {
-        if let Meta::List(MetaList {
-            ref path,
-            ref nested,
-            ..
-        }) = attr
-            .parse_meta()
-            .unwrap_or_else(|_| panic!("Attribute {attr:?} cannot be parsed"))
-        {
-            if path
-                == &parse_str("pool_mutating_action").expect("pool_mutating_action is valid path")
-            {
-                for nested_meta in nested.iter() {
-                    if let NestedMeta::Lit(Lit::Str(litstr)) = nested_meta {
-                        index = Some(i);
-                        return_value =
-                            Some(parse_str::<Ident>(&litstr.value()).unwrap_or_else(|_| {
-                                panic!("{} is not a valid identifier", litstr.value())
-                            }));
-                    } else {
-                        panic!("pool_mutating_action attribute must be in form #[pool_mutating_action(\"REJECTION LEVEL\")]");
-                    }
+    if let Some((index, return_value)) = attrs
+        .iter()
+        .enumerate()
+        .filter(|(_, a)| a.path().is_ident("pool_mutating_action"))
+        .map(|(i, a)| {
+            if let Meta::List(MetaList { ref tokens, .. }) = a.meta {
+                if let TokenTree::Literal(litstr) = tokens.clone().into_iter().next().expect("stratisd code always supplies string literal argument to pool_mutating_actions") {
+                    (
+                        i,
+                        parse_str::<Ident>(litstr.to_string().trim_matches('"')).unwrap_or_else(|_| { panic!("{} is expected to be a valid identifier", litstr.to_string().trim_matches('"'))})
+                    )
+                } else {
+                    panic!("pool_mutating_action attribute must be in form #[pool_mutating_action(\"REJECTION LEVEL\")]")
                 }
+            } else {
+                panic!("pool_mutating_action attribute must be in form #[pool_mutating_action(\"REJECTION LEVEL\")]")
             }
+        })
+        .last() {
```

Comment:
What is the rationale for using last? Is it in order to preserve the same behavior as before where the last value would be the last value found in the loop? If so, I might request that we change this to ensure there is exactly one and use that value or return None.

Reply:
That is the rationale. I was trying to avoid avoidable behavior changes. But I'm all for changing the behavior, so I'll do that while I'm here.

Reply:
I went with panic if encountered twice, I think that's more appropriate.